File: Python/주민경/[Week 1]자료구조/[PGS] 올바른 괄호_LV2.py
def solution(s):
    answer = False
    stack = [n for n in s]
    openStack = []

    for k in stack:
        if k == "(":
            openStack.append("(")
        else:
            if openStack:
                openStack.pop()
            else:
                return answer
    
    if not openStack:
        answer = True
    
    return answer

File: Java/양채연/[Week 1]자료구조/[PGS] 올바른 괄호_LV2.java
import java.io.*;
import java.util.*;
public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		String s = br.readLine();
		bw.write(String.valueOf(solution(s)));
		bw.flush();
		bw.close();
	}
	public static boolean solution(String s) {
		Stack<Character> stack = new Stack<>();
		for (int i = 0; i < s.length(); i++) {
			if (s.charAt(i) == '(') {
				stack.push(s.charAt(i));
			} else {
				if (stack.isEmpty())
					return false;
				else
					stack.pop();
			}
		}
		return stack.isEmpty();
	}
}